bin/polykey.ts

```diff
@@ -0,0 +1,373 @@
+#!/usr/bin/env node
+
+// this will be a CLI entrypoint to this library
+// we expect that this can be used as a CLI
+// application directly as js-polykey
+
+import { program } from 'commander'
+import inquirer from 'inquirer'
+import PolyKey from '../src/Polykey'
+import chalk from 'chalk'
+import os from 'os'
+import fs from 'fs'
+
+let pk: PolyKey
+// Write a `file`
+let tempDir: string
+/*******************************************/
+// initialization
+function initPolyKey(
+    password: string | null,
+    polykeyPath: string | undefined = undefined,
+    publicKeyPath: string | undefined = undefined,
+    privateKeyPath: string | undefined = undefined,
+    privatePassphrase: string | undefined = undefined,
+    verbose: boolean | undefined = undefined
+) {
+    try {
+        // Initialize polykey
+        if (password !== null) {
+            pk = new PolyKey(Buffer.from(password), undefined, undefined, polykeyPath)
+        } else if (process.env.PK_KEY !== undefined) {
+            pk = new PolyKey(Buffer.from(process.env.PK_KEY!), undefined, undefined, polykeyPath)
+        } else {
+            console.log(chalk.red('failed to initialize polykey: password was null'));
+        }
+        // import keys if provided
+        if (publicKeyPath !== undefined) {
+            pk.km.loadPublicKey(publicKeyPath)
+            if (verbose) {
+                console.log(chalk.green('public key successfully imported'))
+            }
+        }
+        if (privateKeyPath !== undefined) {
+            if (privatePassphrase !== undefined) {
+                try {
+                    pk.km.loadPrivateKey(privateKeyPath)
+                    if (verbose) {
+                        console.log(chalk.green('private key successfully imported'))
+                    }
+                } catch (err) {
```

Comment:
For the application entrypoint, you should be doing it similar to in Loja. In Loja we have 1 catch all exception handler that is the root exception handler. At that point, the program should exit with a fatal error and report it.

All other exceptions should be caught and handled. Also it doesn't make sense to handle it only by logging out an error. That should only happen at the root exception handler. Because any other exception handler should actually **handle** the error so there's no more error!

See loja's server/index.ts.

Reply:
I have created a root exception handler now.